pipeline/transformer/transformer_marc_common/src/main/scala/weco/pipeline/transformer/marc_common/transformers/MarcElectronicResources.scala

```diff
@@ -0,0 +1,143 @@
+package weco.pipeline.transformer.marc_common.transformers
+
+import grizzled.slf4j.Logging
+import weco.catalogue.internal_model.identifiers.IdState
+import weco.catalogue.internal_model.locations.AccessStatus.LicensedResources
+import weco.catalogue.internal_model.locations.{
+  AccessCondition,
+  AccessMethod,
+  AccessStatus,
+  DigitalLocation
+}
+import weco.catalogue.internal_model.locations.LocationType.OnlineResource
+import weco.catalogue.internal_model.work.Item
+import weco.pipeline.transformer.marc_common.logging.LoggingContext
+import weco.pipeline.transformer.marc_common.models.{
+  MarcField,
+  MarcRecord,
+  MarcSubfield
+}
+
+import java.net.URL
+import scala.util.{Failure, Success, Try}
+
+trait MarcElectronicResources extends Logging {
+
+  protected def toItems(
+    record: MarcRecord
+  )(implicit ctx: LoggingContext): Seq[Item[IdState.Unminted]] =
+    record
+      .fieldsWithTags("856")
+      .flatMap(field => toItem(field))
+
+  protected def getLabel(field: MarcField): String =
+    field.subfields.filter(_.tag == "y").map(_.content).mkString(" ")
+
+  // Return title as Left or linkText as Right
+  protected def getTitleOrLinkText(
+    field: MarcField
+  ): Either[String, String] = Left(getLabel(field))
```

Comment:
Does this need a path for `Right` too?

Reply:
Not yet.

The left/right thing replaces a previous 2-tuple in the Sierra transformer that decided whether to put this label in the title or the linkText based on whether the label contained an imperative verb. All the labels in the sample data appear to be the name of the resource, so they are all "titles". (Actually, according to MARC, the label is the Link Text, but since we have imperative link text, I'm calling it a title)

This is based on a historic quirk of the existing catalogue, where the label has been used for multiple purposes. As EBSCO are selling this data, I expect it will be consistent.

When we start working with actual data, we may find that we need distinct Left and Right paths in the EBSCO